### ISSUE TITLE
Add FreeBSD support to electron

### ISSUE THREAD

**Issue by teran-mckinney** [atom/electron#3797]
[![Bountysource](https://api.bountysource.com/badge/issue?issue_id=29075473)](https://www.bountysource.com/issues/29075473-add-freebsd-support-to-electron?utm_source=29075473&utm_medium=shield&utm_campaign=ISSUE_BADGE)

Related: #2452

**Comment by zcbenz** [comment]
As mentioned in #2452, currently we don't have any plan to add support for FreeBSD, but if anyone is interested in sending a pull request, we will be very happy to accept it.

**Comment by teran-mckinney** [comment]
Hi @zcbenz,

Could you please keep this open? I opened the request and added a bounty to encourage someone to send in a PR adding in FreeBSD support.

Thank you,
Teran

**Comment by zcbenz** [comment]
Okay.

**Comment by yonas** [comment]
I'm getting close to compiling electron on FreeBSD 10.3.

I have two errors standing in my way:

**Comment by yonas** [comment]
I'm getting close to compiling electron on FreeBSD 10.3.

I have two errors standing in my way:

https://gist.github.com/yonas/06d08aa591e4a452b16c1e9a9ac5e0f3

but I'm not sure how to fix them. Any C/C++ gurus here? :)

**Comment by zacts** [comment]
I see that $200 USD has been added to the bounty? Is this going to happen? I think it will rock if it does.

**Comment by zacts** [comment]
+1

**Comment by yonas** [comment]
Sorry, I'll try to spend more time on this. If anyone knows C/C++ very well, there are Linux specific parts that need to be re-implemented for FreeBSD.

Yes, Electron is becoming more popular and being used in more apps that just the Atom editor. I'd love to have the [Black Screen](https://github.com/shockone/black-screen) terminal running on my FreeBSD machine.

**Comment by prash-wghats** [comment]
Was able to build electron and VSCode in freebsd11RC1. the build instruction is posted in
https://gist.github.com/prash-wghats/89be1ee069d2acf23c289e9c606616e1

**Comment by yonas** [comment]
@prash-wghats Great work!!! I'm curious, why did you compile with VSCode? Is there anything that GCC or CLANG is missing?

**Comment by ilundhild** [comment]
This is awesome.  Can't wait to try it out, and would love to see those changes moved forward.

**Comment by enlight** [comment]
@yonas @prash-wghats didn't build **using** VSCode, he built Electron & VSCode.

**Comment by prash-wghats** [comment]
@yonas Thanks. Built electron using clang, then used it to run VSCode.

**Comment by grahamperrin** [comment]
Upped the bounty. 

(Incidentally: the reference, above, from an _apparently_ Windows-related issue was the result of steering away from a multi-platform issue for the same product.)

**Comment by patrickalima98** [comment]
Hello electron team, my compliments!!!! First I would like to congratulate you on the project and also leave here my desire as well as many others to see the electron running in the FresBSD, your application is very impressive and it would be great if you could add support to FreeBSD I very much enjoyed the platform and it would enable the BSDS systems to leverage enabling the creation of incredible apps. I will do my best to bring other people into this petition. :-)

**Comment by nomadlogic** [comment]
Any chances of the patches referenced by prash-wghats commented on Oct 12, 2016 will be incorporated upstream?  Are their blockers preventing this from happening, or anything the FreeBSD community can do to usher this along?

**Comment by denim2x** [comment]
@teran-mckinney It'd be much easier to use [Linux Binary Compatibility](https://www.freebsd.org/doc/handbook/linuxemu.html) since prebuilt binaries for Linux are already available.

With a custom [electron-packager](https://github.com/electron-userland/electron-packager) you'd be able to bundle your application with all binary dependencies on Linux and execute it on FreeBSD via a special shell script; what's your opinion?

**Comment by prash-wghats** [comment]
https://github.com/prash-wghats/VSCode-For-FreeBSD/releases. 
This binaries were built with freebsd 11.

**Comment by yzgyyang** [comment]
Soooo~ How's everything going? I am more than happy to know if electron is finally available on FreeBSD, or if I can help with something.

**Comment by f1ana** [comment]
I'm also curious what progress, if any, has been made on this.  I've been periodically checking but not really seeing any updates.  I doubt that I have the know-how to contribute directly, but I'm more than happy to add to the bounty should someone commit to the work.

**Comment by grahamperrin** [comment]
> … if electron is finally available on FreeBSD … 

From the Bountysource page: 

>> … expects to finish by Sep 19, 2017. Will try to take a crack at this.

**Comment by ddavidebor** [comment]
This is absolutely great. I will try to help with bug fixes after the initial implementation is done.

**Comment by RainbowHackerHorse** [comment]
^ likewise. 
Gonna keep a close eye on this,and help however I can. 
Been working on porting some electron apps to the Linuxulator, but native support is best.

**Comment by gevorgvoskanyan** [comment]
Any progress on getting a PR ready? I'd be happy to offer some C++ help if needed.

**Comment by yzgyyang** [comment]
Do we have any news on this recently?

**Comment by sfourman** [comment]
I would really like FreeBSD support

**Comment by stites** [comment]
touche -- bumped the bounty as well.

**Comment by dijitaltrix** [comment]
Are there any updates?

**Comment by ararslan** [comment]
I just added to the bounty as well. Hopefully someone isn't just sitting on a solution and waiting to get as much for a bounty as possible. 😄

**Comment by ddavidebor** [comment]
Well one guy arrived at a working build the last year, but additional work
is undoubtedly required.

Il giorno mer 29 nov 2017 alle ore 20:07 Alex Arslan <
notifications@github.com> ha scritto:

**Comment by stites** [comment]
lols, @ararslan -- I'm not sure there are enough FreeBSD users out there to take advantage of. Just the 19+ people who watch this thread : P

**Comment by emaste** [comment]
FreeBSD is in the 4th spot for desktop OS marketshare!

**Comment by ararslan** [comment]
All the more reason to have Electron support. 🙂

**Comment by brenca** [comment]
> FreeBSD is in the 4th spot for desktop OS marketshare!
[Indeed](https://www.netmarketshare.com/operating-system-market-share.aspx?qprid=9&qpcustomb=0), or in other words, at 0.00%.

**Comment by yonas** [comment]
@stites I think there are plenty of FreeBSD, PC-BSD, OpenBSD, and NetBSD users who want to use Electron - and all the apps, like Atom, that are built on it - but they aren't all on this thread.

**Comment by brenca** [comment]
BTW does chromium support FreeBSD?

**Comment by ararslan** [comment]
[Sort of](https://wiki.freebsd.org/Chromium)

**Comment by brenca** [comment]
@ararslan Well, just based on this, adding support in electron would require a fair few of patches, which would be a large burden on those of the maintainers who do the chromium upgrades.

**Comment by yonas** [comment]
@brenca The first thing is to figure out what patches are required to make Electron run properly on FreeBSD.

After that, FreeBSD has a tradition of maintaining its own patches in the "Ports Collection". Some patches will be easy for the upstream maintainer to absorb, and so those patches will be committed upstream over time. That is how we've worked with all other open source communities.

**Comment by brenca** [comment]
That would be superb, I'm just a bit pessimistic, that's all. :)

**Comment by ddavidebor** [comment]
It's not just about the desktop users. Electron on BSD would allow it to be used in commercial embedded hardware application.

Currently QT has basically a market monopoly. 

Electron, together with mobile frameworks such as [ionic](https://github.com/paulsutherland/Polyonic) has the power to become the leading embedded UI platform in the coming years.

**Comment by YurySolovyov** [comment]
Are there CIs who can run BSD?

**Comment by ararslan** [comment]
No major services (it was requested of Travis and they effectively said "not now"), though some projects roll their own. For example, [Julia](https://github.com/JuliaLang/julia) has Buildbot-powered FreeBSD CI running on hardware at National Chiao Tung University in Taiwan. I've wanted to start my own service, but that takes 💰, so in the mean time I think projects need to find their own ways.

**Comment by YurySolovyov** [comment]
I wonder if you can build+run tests of electron inside docker on travis or other CI hosting...

**Comment by ddavidebor** [comment]
> I wonder if you can build+run tests of electron inside docker on travis or other CI hosting...

Wonder no more. You can't.

**Comment by prash-wghats** [comment]
Updated Vscode (v1.17.2) and Electron (1.7.7) for freebsd. 
https://github.com/prash-wghats/VSCode-For-FreeBSD/releases

**Comment by yonas** [comment]
@prash-wghats I was confused about why we needed to build VSCode the first time I saw your repo. After reading your previous comments, it seems we don't need to build VSCode at all.

[Would a build script like this work?](https://github.com/yonas/VSCode-For-FreeBSD/blob/master/build-electron.sh)

If that's true, we should just [create a port](https://www.freebsd.org/doc/en_US.ISO8859-1/books/porters-handbook/porting-submitting.html) using [this form](https://bugs.freebsd.org/bugzilla/enter_bug.cgi?product=Ports%20%26%20Packages&component=Individual%20Port%28s%29).

**Comment by prash-wghats** [comment]
Yes , the script will build electron.  The electron binaries will be placed in electron/dist/* folder. I have not created a freebsd port before. If you can do it please go ahead.

**Comment by prash-wghats** [comment]
The script builds libchromiumcontent library from the FreeBSD port of chromium, then uses that to build electron.

**Comment by yonas** [comment]
@prash-wghats Awesome, I'll try to get this into the ports collection.

If you have time, would you be able to try building [Atom](https://github.com/atom/atom)? I know a lot of people would love to have that editor in FreeBSD.

**Comment by prash-wghats** [comment]
I will give it a try

**Comment by yonas** [comment]
@prash-wghats Thanks, much appreciated!

**Comment by RainbowHackerHorse** [comment]
Happy to contribute or help get it into ports if needed :)
Glad to see movement on this!

**Comment by yonas** [comment]
I'm happy to say that, with the help of @prash-wghats, I was also able to get electron to run on FreeBSD 11.1. :beer: 

I'll try to get this into ports asap.

**Comment by yzgyyang** [comment]
Congrats! Happy to see such progress on this, so excited that thousands of new apps are coming to FreeBSD. Also please let me know how I could help.

**Comment by dariusc93** [comment]
@yzgyyang dont get to happy just yet. It is a jump, but you still are at the mercy of many of the developers that build around electron to make sure their apps have support for freebsd. Some may just be a seamless transition, some may require additional modifications to their source, and others may require a rewrite (hopefully that wont be the case)

**Comment by yonas** [comment]
@dariusc93 Can you give an example of an Electron app that has OS-specific functionality?

My understanding is that isn't possible.

**Comment by dariusc93** [comment]
@yonas You are right. They are like websites, but with electron you can also make use of os specific things as well just like any node app. Some apps  load shared libraries that are built for a specific platform and those libraries would have to be ported to freebsd. I cannot think of the apps that done this from the top of my head as I stop using alot of electron app because of this as they may target windows or macos but not linux (and soon to be freebsd)

**Comment by prash-wghats** [comment]
@yonas like node modules NSFW, minidump, spellchecker ... Some of this is trivial like adding OS name in the configuration file. others like nsfw will require platform specific code.

**Comment by yonas** [comment]
@dariusc93 @prash-wghats Ah, thanks. Hopefully we don't run into too many of those :)

**Comment by dariusc93** [comment]
@prash-wghats what do you believe is needed now for things to get brought upstream?

**Comment by prash-wghats** [comment]
@dariusc93 For electron, need to make a port package and submit i guess, because all the work has been already done by the chromium port for freebsd. As for apps built by electron like vscode or atom, needs Freebsd support in some of the node modules.

**Comment by dariusc93** [comment]
Making a port package isnt the problem I dont believe. I mean making a pr towards electron (and any of its dependencies, if required) so it would reduce patches in the port, and so if someone wishes to build from git they would have the option of doing that.

**Comment by prash-wghats** [comment]
you can try Atom and LightTable for freebsd.
https://github.com/prash-wghats/Electron-VSCode-Atom-For-FreeBSD
https://github.com/prash-wghats/Electron-VSCode-Atom-For-FreeBSD/releases/tag/v123.081

**Comment by yonas** [comment]
Thanks @prash-wghats !! Atom finally runs on FreeBSD :-)

**Comment by ararslan** [comment]
I believe @prash-wghats should collect the bounty, or at least once Electron is available in Ports. At what point can this be considered "solved"? When the requisite patches have been upstreamed such that Electron can be built from source on FreeBSD outside of the Ports Collection, or once it's available in Ports?

**Comment by yonas** [comment]
I also support get @prash-wghats getting the bounty. Amazing work.

**Comment by dariusc93** [comment]
I do believe @prash-wghats should get the bounty. He has put work into getting working and once things are push upstream to the repo, other devs could simply add support for freebsd with little to no excuses.

**Comment by yzgyyang** [comment]
Is anyone making a new port for Electron now? I didn't see any PR on bugs.freebsd.org. I would like to do it if no one is up to it right now.

**Comment by yonas** [comment]
@yzgyyang I was going to, but don't have time right now. If you could create the PR, that'd be awesome. Thanks!

**Comment by yzgyyang** [comment]
@yonas Thanks! Can I see your progress somewhere so I can continue on that? Is it at yonas/VSCode-For-FreeBSD? (I can see some build scripts there, not sure which is the right one to use)

**Comment by yonas** [comment]
@yzgyyang Yes, https://github.com/yonas/VSCode-For-FreeBSD has the changes needed to fix some compile errors.

**Comment by yzgyyang** [comment]
@yonas Alright then, I will try building it following your changes, and will start making a new port if everything works.

**Comment by yzgyyang** [comment]
With @prash-wghats 's work and @yonas 's patches, and some more patches to fix build errors, Electron can be successfully built and run on my machine. I am already working on making a new port for it.
![screenshot_2018-01-11_18-21-49](https://user-images.githubusercontent.com/16748524/34852594-6fe1a15c-f6fd-11e7-8207-042c001f5fa9.png)

**Comment by dariusc93** [comment]
My question is will some of the patches be brought upstream? Doing so can reduce the patches in the ports, and make upgrading easier in the future vs having to update the patches.

**Comment by yzgyyang** [comment]
@dariusc93 IMO we definitely should submit those patches to upstream after ensuring Electron is fully functional on FreeBSD. We probably still need to take care of them until FreeBSD is "officially supported" upstream though.

**Comment by herrbischoff** [comment]
@yzgyyang: As stated [further above](https://github.com/electron/electron/issues/3797#issuecomment-164410644) it does not appear likely the project will officially support FreeBSD. Maybe one of the core team could make a clear statement about whether it will be maintained or not once a working pull request has landed.

**Comment by ddavidebor** [comment]
Got to start somewhere @herrbischoff

**Comment by monty11ez** [comment]
Any updates on the port? @yzgyyang or @prash-wghats

**Comment by yzgyyang** [comment]
@monty11ez I'm on it. Continued working with @prash-wghats and my mentors to update the build script and test the port, but the port still needs lots of investigation. In the meantime, please feel free to use the latest build script that @prash-wghats provided, it worked great.

**Comment by dszidi** [comment]
@yzgyyang What version of FreeBSD are you targeting? 11 or 12?

**Comment by Alhadis** [comment]
Great. Now do the same for OpenBSD.

@qbit, who maintains OpenBSD's port of Node.js, tried building an Electron port last year.

[Poor **@qbit** suffered terribly](https://deftly.net/posts/2017-06-01-measuring-the-weight-of-an-electron.html).

**Comment by dariusc93** [comment]
@dszidi while I cannot speak directly for trueos, but electron should be able to be built if youre using the right dependencies. Could you post any build logs?

**Comment by ckerr** [comment]
A little late to this ticket, but to put in my two cents:

I don't think it makes sense for the GitHub Electron team to officially support BSD because none of us are running a (non-macOS) BSD and we're generally overloaded with the platforms we *do* support. So it's not anything about BSD and all about resources.

That said, I love that there is community interest in this and, as @zcbenz said above, we'd welcome patches for this and I'd be happy to help get PRs landed.

One comment though: I notice that the screenshot @yzgyyang posted has 1.6.x running in it. Merging will go a lot more smoothly if the PR is against master instead :smile:

**Comment by Alhadis** [comment]
@ckerr From what I gather, building Electron is both nightmarish and extremely convoluted. Even getting one version to build on OpenBSD should be enough for somebody to maintain a port for it.

The fact that both Chromium and Node have actively-maintained OpenBSD ports suggests that Electron's build process has room for improvement.

**Comment by yzgyyang** [comment]
> What version of FreeBSD is your port targeting? 11 or 12? I'm trying to build electron build script on TrueOS (FreeBSD 12) and am having trouble.

@dszidi Currently FreeBSD 11.1 is hard-coded in our patches and WIP ports. It would be a simple code change, but 11.1 is what we are working on right now. See https://github.com/prash-wghats/Electron-VSCode-Atom-For-FreeBSD/issues/8.

> we'd welcome patches for this and I'd be happy to help get PRs landed.

@ckerr Thanks! I believe that is our next goal.

> I notice that the screenshot @yzgyyang posted has 1.6.x running in it. Merging will go a lot more smoothly if the PR is against master instead 😄

@ckerr Our ongoing work has been shifted to build against 1.8 (latest stable) version.

**Comment by yzgyyang** [comment]
An update: Electron 1.8.2 Beta 2 can be built and installed with the ports system! There's still a lot of work to bring this to the ports tree (more patches, more testing, switch to use 1.8.2 stable, etc. etc.), so no estimated time for now.
![screenshot_2018-03-01_13-55-08](https://user-images.githubusercontent.com/16748524/36864141-d2a97ea4-1d59-11e8-819f-901a6b120007.png)

**Comment by mmokhi** [comment]
@yzgyyang Hey Congrats 😃 
I and @pizzamig had some efforts in summer (lead to having gyp into tree) maybe it can help you to reach to destination faster 🙂 
On https://github.com/pizzamig/freebsd-ports you can find a branch
called devel-libchromiumcontent that contains the initial work on some of dependencies and libraries.
Hope it helps and thanks in advance for efforts 👍

**Comment by pizzamig** [comment]
@yzgyyang Hey! Great work!
I can help migrating the "build_electron.sh" script in a FreeBSD port compatible way

**Comment by yzgyyang** [comment]
@pizzamig A lot of work was done by @prash-wghats on a buildable build_electron.sh, and I am already on writing ports and testing (check this out: https://github.com/yzgyyang/freebsd-ports-electron). More eyes and hands are always welcome 😄

**Comment by pizzamig** [comment]
great, I'll do some pull request in the weekend, there're some stuff to be fixed, but it's really promising

**Comment by ckerr** [comment]
Should've commented on this sooner, but better late than never:

@Alhadis simplifying the build process would benefit BSD in exactly the same way that it would benefit all platforms, so maybe that should be split off to a separate issue if you have specific improvement suggestions to make. Thanks!

**Comment by sfourman** [comment]
Has anyone successfully made a FreeBSD port of this work yet?

**Comment by Alhadis** [comment]
Weren't you off having fun before enjoying having Electron-support on FreeBSD? ;-)

**Comment by wezm** [comment]
FYI I've added $50 to [the bounty on this issue](https://www.bountysource.com/issues/29075473-add-freebsd-support-to-electron), which brings it to $750.

**Comment by pizzamig** [comment]
Here you can find the FreeBSD port that will build and install electron on FreeBSD:
https://github.com/yzgyyang/freebsd-ports-electron

Currently, the port is poudriere compatible, but not yet committed; in other words, you can build it, but there's no package yet.
Please, feel free to use it and test it! We need your feedback

**Comment by duncan-bayne** [comment]
@pizzamig Great, thanks!  Will test soon.  Hoping to get Cypress running on FreeBSD, so this is great timing :)

**Comment by RainbowHackerHorse** [comment]
@fabianfreyer I think this is a fantastic idea, considering the large number of ports we'll see coming into the tree once this is committed.

@yzgyyang is there a holdup on committing, or are you just testing first? If you submitted and are waiting on a committer, I can ping my mentor and see if she can poke at it, or if someone else I know with a commit bit can :)

**Comment by thenexxuz** [comment]
Is there any update on adding BSD support for electron apps?

**Comment by thenexxuz** [comment]
BTW, the bounty is now $800.00

**Comment by udp** [comment]
Why is this closed?  I'm happily using both Chromium and node.js on FreeBSD, so there's no technical reason Electron shouldn't be able to work here.

**Comment by udp** [comment]
Oops, sorry, saw the "closed" flag for the Haiku OS post above and thought it was for this issue. -_-

**Comment by prash-wghats** [comment]
Electron is working in FreeBSD. check https://github.com/yzgyyang/freebsd-ports-electron or for prebuilt binaries https://github.com/prash-wghats/Electron-VSCode-Atom-For-FreeBSD

**Comment by ararslan** [comment]
> Electron is working in FreeBSD.

The binaries are great, and I've used your VSCode binaries a bit on 11.2. It would be great to get Electron and friends into the Ports Collection though. I think that's what would effectively "complete" this issue.

**Comment by asaf-romano** [comment]
Added 50$ to bounty.

**Comment by grahamperrin** [comment]
> … vscode and the actual electron binary are working on TrueOS/FreeBSD 12 (haven't tried Atom yet). …

### Atom 1.23.0

`Atom-v1.23.0_5_6.tar.xz` from https://github.com/prash-wghats/Electron-VSCode-Atom-For-FreeBSD/releases/tag/atom_v1.23.0

```text
$ date ; uname -v ; pwd
Sun 30 Sep 2018 11:31:46 BST
FreeBSD 12.0-ALPHA7 r338956 GENERIC-NODEBUG 
/home/grahamperrin
$ Applications/Atom/atom Desktop/test.txt 
Fontconfig warning: "/usr/local/etc/fonts/local.conf", line 1093: saw number, expected matrix
Attempting to call a function in a renderer window that has been closed or released.
Function provided here: worker.js:61:22
Remote event names: destroyed, crashed
$ 
```

The three lines after the (negligible) `fontconfig` warning occur when quitting the application.

**Comment by pizzamig** [comment]
https://github.com/yzgyyang/freebsd-ports-electron is a port, compatible and usable with poudriere, to build a FreeBSD package and is largely based on the great work of @prash-wghats . Without his insight, nothing would be possible.
Unfortunately, the port, as it is, cannot land to the Port Collection, it has too many tricks and things not allowed (especially the libcc and the node packages).
Currently, I'm working on an acceptable port for libcc. If you have ever build chromium from sources, you'll now why it's "time consuming".

**Comment by nornagon** [comment]
libcc is deprecated and not used in the current build of electron, so probably not a great idea to work on a port for it :)

**Comment by thenexxuz** [comment]
Having it work in ports is good and all and I agree that it is a good start but there are apps that are closed source (GitKraken for example) I would not expect to have to directly support BSD but if it is an official platform for electron then they could cross compile a BSD build. But that cannot happen until there is an official build. That's just my opinion.

**Comment by dariusc93** [comment]
I think the main hurdle currently to bring the port upstream is the chromium library. Could be wrong though, but other than that, if the patches for electron itself is brought upstream, then it should be good to go from there but there is also more to it since there is likely libraries out there that isnt built to run on BSD or isnt configured for the platform. It would still need to be work for devs to make sure any native modules can run on BSD as well. If there isnt any dependencies that make use of native modules, then it should be fine.

**Comment by kreinholz** [comment]
+1 for this! Would love to get Atom and/or VS Code running on FreeBSD. CodeLite and Eclipse are all well and good, but I'd really like to be able to develop with Atom or VS Code without having to hop into a virtual machine running Linux.

**Comment by thenexxuz** [comment]
How high does the bounty need to get before someone decides to do this? $1000 USD?

**Comment by tagattie** [comment]
Hi, I have made an experimental port of electron 4.1.x and 5.0.0 for FreeBSD. This is largely based on the works by @prash-wghats and FreeBSD chromium team. If you are interested, take a look at https://github.com/tagattie/FreeBSD-Electron.

I'm running VS Code on FreeBSD with it for a while.

**Comment by nornagon** [comment]
@tagattie looks like that includes quite substantial changes to Chromium. Are you working with the Chromium team at all to upstream those changes? If not I'm afraid they will quickly go out of date :(

**Comment by tehpeh** [comment]
@nornagon the Readme states the patches come from the Chromium team.

**Comment by pizzamig** [comment]
@nornagon The FreeBSD Chromium team maintains a large set of patches, because Google doesn't accept those changes, because FreeBSD (and any other BSDs) is not a platform they're interested in.

**Comment by apfeifer** [comment]
Published a few hours ago.
https://www.freshports.org/devel/electron4/

**Comment by homerstrong** [comment]
Ryan Ballinger 308 elm st. Homer Michigan 49245 there is a bug issue in my program my wife/partner is not supportive no one can expect someone to get any better when there Spouse is acting aggressive and not loving like a spouse should be not asking for a slave just the old goofy girlfriend i had or this will never work

**Comment by thenexxuz** [comment]
Not sure previous message should be removed because of personal info in the message.

**Comment by dariusc93** [comment]
I feel like it should be removed

**Comment by kreinholz** [comment]
Agreed. It doesn't appear to be related to this thread's topic, and that level of personal info shouldn't be left in the wild.

**Comment by zoujiaqing** [comment]
Thanks, can upgrade it to 6.0.x?

**Comment by lapo-luchini** [comment]
There's a work-in-progress [branch for 5.x](https://github.com/tagattie/FreeBSD-Electron/tree/5-0-x) but nothing on 6.x so far as far as I can see.

**Comment by ararslan** [comment]
If this is now a registered port, someone should collect the BountySource bounty.

**Comment by thenexxuz** [comment]
The point of this, if I understand correctly, to have Electron be able to generate a BSD build from any other platform. For example, on a Mac you can generate a Windows EXE or a Linux Binary. This is to allow them to generate the BSD Binary, not having to run BSD and then compile from there.

**Comment by crazyBaboon** [comment]
+1 - I would love to  have **atom** available on FreeBSD

**Comment by grahamperrin** [comment]
Might future resolution of this issue allow me to (for example) install **FF Password Exporter** https://github.com/kspearrin/ff-password-exporter on FreeBSD-CURRENT? 

From https://pastebin.com/Z3mmYLh9 I imagine not. Line 19: 

> Error: GET https://github.com/electron/electron/releases/download/v3.0.6/electron-v3.0.6-freebsd-x64.zip returned 404

Background: https://github.com/MrAlex94/Waterfox/issues/1258#issuecomment-552139342

**Comment by chavez243ca** [comment]
Want to see this happen so they can bring Brim to FreeBSD! Will try to contrib to the bounty.

**Comment by pizzamig** [comment]
electron has been available on FreeBSD for a while.
The maintainer had some issues the packages have been removed when python 2.7 (a dependency) have been removed.
Now he is back and he's working on restoring them (electron7, electron8 and electron9)

**Comment by thenexxuz** [comment]
> electron has been available on FreeBSD for a while.

While you are able to RUN electron apps in FreeBSD you cannot BUILD an app and target BSD like you can with the other OSs. I write an app on a Mac and make a build target Linux or Windows, the same is not true for BSD. In order for a BSD build for an app to be made it needs to be built IN BSD. THAT is the support needed.

**Comment by tcdude** [comment]
I upped the bounty on this a bit.. not sure if it will help spark some more interest in this...

**Comment by michaellee8** [comment]
Are anyone working on this issue yet? This one looks like interesting to me, especially with the bounty. So is the requirement to have FreeBSD as a target aviliable?

**Comment by kreinholz** [comment]
I think this issue should be closed, as electron12 has arrived in the Ports tree:

https://www.freshports.org/devel/electron12/

I'm personally running electron v.12.0.7 on FreeBSD 13.0 amd64, and vscode v.1.45.0 at least builds and runs just fine on this system. I can't speak to other architectures, but on amd64 at least Electron on FreeBSD is a reality.

**Comment by michaellee8** [comment]
> I think this issue should be closed, as electron12 has arrived in the Ports tree:
> 
> https://www.freshports.org/devel/electron12/
> 
> I'm personally running electron v.12.0.7 on FreeBSD 13.0 amd64, and vscode v.1.45.0 at least builds and runs just fine on this system. I can't speak to other architectures, but on amd64 at least Electron on FreeBSD is a reality.

I believe this should be closed then, maybe the author of the freebsd patch himself should go and collect his well deserved bounty

**Comment by emaste** [comment]
There are currently 996 patch files in the FreeBSD devel/electron12 port; it would be very beneficial for long-term maintenance to get as much of this upstreamed as possible.

**Comment by thenexxuz** [comment]
I personally do not think it should be closed because you cannot build on one of the other OSs targeting BSD, which I believe was the original intent of this, and the reason why I contributed to the bounty.

**Comment by dariusc93** [comment]
What would be needed to start getting some patches brought upstream?

**Comment by dmilith** [comment]
Would be great to get official support from the project for 4th biggest open-source system…

**Comment by probonopd** [comment]
The lack of official FreeBSD support in Electron seems to be holding back an official Session client from FreeBSD.
It gives the (incorrect?) impression that "Electron is not supported on FreeBSD", even though there is a [port](https://www.freshports.org/devel/electron12) which suggests that it technically works.

**Comment by thenexxuz** [comment]
Remember the point of this issue is not to be able to build Electron on BSD, rather the ability to build a BSD version of your app on a different platform.

On MacOS you can build a Windows version and a Linux version of your app.
On Windows you can build a Linux version and a MasOS version of your app.
On Linux you can build a MacOS version and Windows version of your app.

I donated to this so that ANY OS can build a BSD version of their app.

**Comment by xliiv** [comment]
> Remember the point of this issue is not to be able to build Electron on BSD, rather the ability to build a BSD version of your app on a different platform.
> 
> On MacOS you can build a Windows version and a Linux version of your app.
> On Windows you can build a Linux version and a MasOS version of your app.
> On Linux you can build a MacOS version and Windows version of your app.

Could anyone put it into the issue desc, then

**Comment by peepopoggers** [comment]
> Would be great to get official support from the project for 4th biggest open-source system…

It is disappointing to not have support.

**Comment by Andros2310** [comment]
Greetings! There is already a port of Electron for FreeBSD (https://github.com/tagattie/FreeBSD-Electron) by @tagattie

**Comment by probonopd** [comment]
Electron is in FreeBSD Ports now:
https://www.freshports.org/devel/electron13/

But FreeBSD is not supported by Electron in electron-builder etc. if I understand it correctly. Or how does one build an Electron app targeting FreeBSD?

**Comment by TZubiri** [comment]
> Remember the point of this issue is not to be able to build Electron on BSD, rather the ability to build a BSD version of your app on a different platform.
> 
> On MacOS you can build a Windows version and a Linux version of your app. On Windows you can build a Linux version and a MasOS version of your app. On Linux you can build a MacOS version and Windows version of your app.
> 
> I donated to this so that ANY OS can build a BSD version of their app.

How quaint. I don't use electron or FreeBSD, but my understanding was that electron app are just local websites run on an (statically linked?) (chromium?) browser, so the objective here would be to "simply" bundle the html, js and css files and make sure that freebsd opens those with the browser?

Seems more than doable.

**Comment by HolgerJeromin** [comment]
> How quaint. I don't use electron or FreeBSD, but my understanding was that electron app are just local websites run on an (statically linked?) (chromium?) browser, so the objective here would be to "simply" bundle the html, js and css files and make sure that freebsd opens those with the browser?

Electron is much more than a lightweight browser. It has many JS APIs which allows native integration (file access, systray icon, custom API of the integrator, window chrome aka no address bar and more) which must not be available (on purpose) with any random, untrusted website.

**Comment by rilysh** [comment]
I found several people were successfully able to build electron on FreeBSD but pretty much in hacky way, and several third-party build I was able to find, but most of them are quite outdated.

Seems there are no hope at all https://github.com/electron/electron/issues/36325#issuecomment-1312683740, until FreeBSD gets a wide range of users, *well macOS contains a lot of FreeBSD code, yet Electron just never give native support for FreeBSD* and to be honest, this time probably porting GNU/Linux version of Electron for FreeBSD would be \*best\* choice so far.

**Comment by probonopd** [comment]
[Visual Studio Code - Open Source ("Code - OSS")](https://www.freshports.org/editors/vscode/) is a thing. And it works nicely on FreeBSD. It uses https://www.freshports.org/devel/electron18/.

**Comment by rilysh** [comment]
Here, I'm not talking about ports, I know electron port exist for FreeBSD, however, my main intention is officially support thing.

**Comment by foxxy432** [comment]
OMGMGGGMGMGMGMGMGG FREEBSD BSD













































# BSD = OLD

**Comment by duncan-bayne** [comment]
> BSD = OLD

... which means it works and is stable 😛

**Comment by foxxy432** [comment]
haha no not stable

**Comment by foxxy432** [comment]
Here are some known issues with FreeBSD that may cause instability:

    Hardware support: FreeBSD supports a wide range of hardware platforms, but some hardware may not be fully compatible or may have limited support. This can lead to issues such as kernel panics, crashes, or poor performance.
    Kernel bugs: Like any other operating system, FreeBSD has bugs in its kernel that can cause instability. These bugs can be caused by errors in code, incorrect configuration, or unexpected interactions between different components.
    Driver issues: Device drivers in FreeBSD can also cause instability if they are not properly maintained or if they have bugs. Outdated drivers or drivers that are not optimized for certain hardware can cause system crashes or malfunctions.
    System configuration: Improper system configuration can also lead to instability in FreeBSD. For example, incorrectly configured network settings or improperly tuned system parameters can cause the system to become unstable.
    Software conflicts: Conflicts between different software packages or libraries can also cause instability in FreeBSD. For instance, conflicting dependencies or version mismatches can lead to crashes or other issues.
    Security vulnerabilities: FreeBSD, like any other operating system, is susceptible to security vulnerabilities. If these vulnerabilities are not addressed promptly, they can cause instability and compromise the security of the system.
    Resource constraints: FreeBSD may require more resources (such as memory or CPU) than other operating systems to run smoothly, especially when running resource-intensive applications. If the system does not have enough resources, it may become unstable or slow down significantly.
    Lack of popularity: FreeBSD is not as widely used as other operating systems, which means that there may be less community support and fewer resources available for troubleshooting and debugging. This can make it more difficult to identify and fix issues quickly, leading to increased instability.
    Development cycle: FreeBSD has a relatively long development cycle compared to other operating systems. This means that new features and bug fixes may take longer to implement, which can result in slower progress in addressing stability issues.

**Comment by duncan-bayne** [comment]
@foxxy432 This isn't the time or place for an OS holy war.

**Comment by yonas** [comment]
@zcbenz

> As mentioned in https://github.com/electron/electron/issues/2452, currently we don't have any plan to add support for FreeBSD, but if anyone is interested in sending a pull request, we will be very happy to accept it.

I've created a PR that should be applied to v22.3.23, not HEAD: https://github.com/electron/electron/pull/39761

**Comment by hhartzer** [comment]
Very cool! Would it be possible to rebase it to main? Quite a lot of work there.

**Comment by Martinfx** [comment]
@yonas do you still work on patches for Electron?
